rust/private/rust.bzl

```diff
@@ -859,15 +884,37 @@ rust_static_library = rule(
         """),
 )
 
+def _rust_shared_library_transition_impl(settings, attr):
+    return {
+        "//command_line_option:platforms": str(attr.platform) if attr.platform else settings["//command_line_option:platforms"],
+    }
+
+_rust_shared_library_transition = transition(
+    implementation = _rust_shared_library_transition_impl,
+    inputs = [
+        "//command_line_option:platforms",
+    ],
+    outputs = [
+        "//command_line_option:platforms",
+    ],
+)
+
 rust_shared_library = rule(
     implementation = _rust_shared_library_impl,
-    attrs = dict(
-        _common_attrs.items() + _experimental_use_cc_common_link_attrs.items() + {
-            "_use_grep_includes": attr.bool(default = True),
-        }.items(),
-    ),
+    attrs = dict(_common_attrs.items() + _experimental_use_cc_common_link_attrs.items() + {
+        "platform": attr.label(
+            doc = "Optional platform to transition the shared library to.",
```

Comment:
Ideally this would also describe how using this attribute interacts with `select` on other attributes: As far as I know, they will only work with Bazel 7+ and will see the "post-transition" platform.

Reply:
Could you elaborate on what you mean?  I'm viewing `rust_shared_library()` and `rust_static_library()` as terminal rules, where a platform transition makes sense, but technically they could be depended on by another binary that's targeting another platform, which would be broken (at least as far as C/Rust style dependencies go, totally fine for bundling).  A thought that comes to mind typing this would be putting the target platform into those providers and then `fail()`ing if they don't match when used as a dependency.  Assuming that's possible 😅

Reply:
I think Fabian is referring to https://github.com/bazelbuild/bazel/issues/15157 which got fixed in Bazel 7.

Reply:
Thanks for providing the reference, @illicitonion. This is what I meant: Before Bazel 7, this may actually break `rust_*` targets with `select` in attributes. Do you happen to have a test for this? I'm not 100% sure what the behavior is in case the transition is a no-op, so maybe this only affects targets with `platform` set.

Reply:
Doesn't look like we can tag `attr`s as `nonconfigurable` unfortunately (I'm guessing that's builtin only), but that seems like it would be the ideal solution here?

Reply:
@illicitonion That's the extent of it as far as I understand, but it does warrant some testing. 

It would be possible to wrap the rules in macros that check the type of `platforms` and `fail` if they see a select, but that's cumbersome.